service/service.go

```diff
@@ -178,26 +178,30 @@ func (s *Service) computeHash() string {
 // saveContext downloads service context to a temp dir.
 func (s *Service) saveContext(r io.Reader) error {
 	var err error
-	s.tempPath, err = s.createTempDir()
+	s.tempPath, err = ioutil.TempDir("", "mesg-")
 	if err != nil {
 		return err
 	}
 
 	s.sendStatus("Receiving service context...", DRunning)
 	defer s.sendStatus("Service context received with success", DDonePositive)
 
-	return archive.Untar(r, s.tempPath, &archive.TarOptions{
-		Compression: archive.Gzip,
-		NoLchown:    true,
-	})
-}
-
-func (s *Service) createTempDir() (path string, err error) {
-	return ioutil.TempDir("", "mesg-"+uuid.NewV4().String())
-}
+	if err := archive.Untar(r, s.tempPath, &archive.TarOptions{
+		NoLchown: true,
```

Comment:
why removing the `Compression: archive.Gzip,` ? Can we still support the `tar.gz` without ?

Reply:
Because Compression is unused while Untar (only works for `archive.Tar(WithOptions)`.

Untar detects compression so any of [uncompressed, bzip2, gzip, xz] can be used (all of them are supported by docker)